modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,161 @@
+
+.Prerequisites
+
+* You must have permission to perform the following actions:
+** Create build configs and start builds.
+** Discover which `SharedSecret` custom resource (CR) instances are available by entering the `oc get sharedsecrets` command and getting a non-empty list back.
+** Determine if a service account is available in your namespace and is allowed to use the given `SharedSecret` CR instance. In other words, you can run `oc adm policy who-can use <identifier_of_specific_SharedSecret>` to see if the desired service account in your namespace is listed.
+
+[NOTE]
+====
+If neither of the last two prerequisites in this list are met, establish, or ask your cluster administrator to establish, the necessary role-based access control (RBAC) so that you can discover `SharedSecret` instances and grant service accounts to use `SharedSecret` instances.
+====
+
+.Procedure
+
+. Create a `SharedSecret` object instance with the cluster's entitlement secret by running the following command:
++
+[IMPORTANT]
+====
+You must have the cluster administrator permissions to create `SharedSecret` objects.
+====
++
+[source,yaml]
+----
+$ oc apply -f -<<EOF
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: etc-pki-entitlement
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+EOF
+----
+
+. Create a `ClusterRole` to grant permission to access the `SharedSecret` object:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: use-share-etc-pki-entitlement
+rules:
+  - apiGroups:
+      - sharedresource.openshift.io
+    resources:
+      - sharedsecrets
+    resourceNames:
+      - etc-pki-entitlement
+    verbs:
+      - use
+----
+
+. Create a `Role` and `RoleBinding` object that grants the Shared Resources CSI driver the permission to access the `SharedSecret` object:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: share-etc-pki-entitlement
+  namespace: openshift-config-managed
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    resourceNames:
+      - etc-pki-entitlement
+    verbs:
+      - get
+      - list
+      - watch
+----
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: share-etc-pki-entitlement
+  namespace: openshift-config-managed
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: share-etc-pki-entitlement
+subjects:
+  - kind: ServiceAccount
+    name: csi-driver-shared-resource
+    namespace: openshift-builds <1>
+----
++
+<1> Enter the name of the namespace where openshift-builds is installed.
+
+. Create a `RoleBinding` object for the ServiceAccount that consumes the `SharedSecret` in your current namespace:
```

Comment:
The `builder` and `pipeline` service accounts are special - these are the service accounts used for BuildConfig builds and Builds for OpenShift builds. I think we should be a bit more specific here:

```suggestion
. Create a `RoleBinding` object for the `builder` and `pipeline` ServiceAccounts in the namespace where builds will run:
```